rpc/src/module/ipc.rs

```diff
@@ -0,0 +1,784 @@
+                *e = meta.clone()
+            }
+        }
+        let snapshot = self.shared.cloned_snapshot();
+        let pipesctx = PipesCtx::default();
+        let script_verifier = TransactionScriptsVerifier::<_, _, Machine>::new_with_generator(
+            Arc::new(tx),
+            dl,
+            snapshot.cloned_consensus(),
+            Arc::new(TxVerifyEnv::new_submit(snapshot.tip_header())),
+            overload_ckb_syscalls,
+            pipesctx.clone(),
+        );
+        std::thread::spawn(move || {
+            // Ignore its return value as it is unimportant.
+            script_verifier.verify_single(
+                match env.script_group_type {
+                    ScriptGroupType::Lock => ckb_script::ScriptGroupType::Lock,
+                    ScriptGroupType::Type => ckb_script::ScriptGroupType::Type,
+                },
+                &env.script_hash.0.pack(),
+                // I need a maximum number of cycles so that the vm can always be stopped.
+                // Not sure if this is a sensible value.
+                snapshot.cloned_consensus().max_block_cycles,
+            )
+        });
+        // In any case, we will close all pipes after a certain period of time.
+        let pipesfin = pipesctx.clone();
+        std::thread::spawn(move || {
+            std::thread::sleep(std::time::Duration::from_secs(8));
+            pipesfin.vw.lock().unwrap().close();
+            pipesfin.vr.lock().unwrap().close();
+            pipesfin.nw.lock().unwrap().close();
+            pipesfin.nr.lock().unwrap().close();
+        });
+        let req = RequestPacket::new(
+            req.version.value() as u8,
+            req.method_id.value(),
+            match fmt {
+                IpcPayloadFormat::Hex => {
+                    let str = req
+                        .payload
+                        .as_str()
+                        .unwrap_or_default()
+                        .strip_prefix("0x")
+                        .unwrap();
+                    let mut buf = vec![0u8; str.len() / 2];
+                    hex_decode(str.as_bytes(), &mut buf)
+                        .map_err(|e| RPCError::custom_with_error(RPCError::IPC, e))?;
+                    buf
+                }
+                IpcPayloadFormat::Json => serde_json::ser::to_vec(&req.payload)
+                    .map_err(|e| RPCError::custom_with_error(RPCError::IPC, e))?,
+            },
+        );
+        pipesctx
+            .nw
+            .lock()
+            .map_err(|e| RPCError::custom_with_error(RPCError::IPC, e))?
+            .write_all(&req.serialize())
+            .map_err(|e| RPCError::custom_with_error(RPCError::IPC, e))?;
+        pipesctx
+            .nw
+            .lock()
+            .map_err(|e| RPCError::custom_with_error(RPCError::IPC, e))?
+            .close();
+        let rep = ResponsePacket::read_from(pipesctx.nr.lock().unwrap().deref_mut()).unwrap();
```

Comment:
How about changing `rep` to `resp`, since `rep` can easily be confused with `req`?